hack/uwm_setup.sh

```diff
@@ -0,0 +1,259 @@
+CONFIGURATION. THE SCRIPT ALSO DEPLOYS A USER APPLICATION IN A NAMESPACE CALLED
+  'ns1'. REVIEW THIS SCRIPT THOROUGHLY TO DETERMINE SUITABILITY FOR YOUR TEST
+                                  ENVIRONMENT
+*******************************************************************************"
+while true; do
+  read -rp "Do you want to continue? (y/n): " YESNO
+  case $YESNO in
+      y ) break;;
+      n ) echo "Exiting."; exit 0;;
+      * ) echo "Please answer 'y' or 'n':";;
+  esac
+done
+echo ""
+}
+
+function kubeadmin_login ()
+{
+echo "Log into your cluster initially as kubeadmin:"; echo ""
+
+# Request cluster API endpoint and kubeadmin token from user:
+read -rp "- Enter cluster API endpoint and port (i.e. https://api.<cluster_name>.<sub_domain>.<domain>:<port>: " APIURL
+echo ""
+unset -v KUBEPW # Make sure the $KUBEPW password variable is not exported
+set +o allexport  # Make sure variables are not automatically exported
+read -rs -p "- Enter kubeadmin token (this will not be echoed to the console and the variable will not be exported): " KUBEPW < /dev/tty &&
+echo ""
+
+# Log in as kubeadmin:
+oc login --token="${KUBEPW}" --server="${APIURL}"
+RESULT=$?
+if [[ "${RESULT}" != "0" ]]; then
+  echo "Login unsuccessful. Exiting."
+  exit 0
+else
+  OCPUSER=$(oc whoami)
+  echo "Now logged in as ${OCPUSER}."
+fi
+echo ""
+}
+
+function enable_uwm ()
+{
+echo "########### Enabling user workload monitoring ##########"
+# Create the `cluster-monitoring-config` ConfigMap if it does not already exist:
+CONFMAPNAME=$(oc get configmap cluster-monitoring-config -n openshift-monitoring 2>/dev/null | grep -s 'cluster-monitoring-config' | awk '{print $1}')
+if [[ "${CONFMAPNAME}" = "cluster-monitoring-config" ]]; then
+  echo "configmap/cluster-monitoring-config already exists in the openshift-monitoring namespace."
+else
+  oc create configmap cluster-monitoring-config -n openshift-monitoring
+fi
+
+# Create a `./tmp/cluster-monitoring-config.yaml` file and append a `data/config.yaml` section that will enable user workload monitoring when applied:
+oc get configmap cluster-monitoring-config -n openshift-monitoring -o yaml > ./tmp/cluster-monitoring-config.yaml
+cat <<EOF >> ./tmp/cluster-monitoring-config.yaml
+data:
+  config.yaml: |
+    enableUserWorkload: true
+EOF
+
+# Apply the `./tmp/cluster-monitoring-config.yaml` configuration to the cluster:
+oc apply -f ./tmp/cluster-monitoring-config.yaml
+
+# List newly created pods in the `openshift-user-workload-monitoring` namespace:
+sleep 60
+oc get pods -n openshift-user-workload-monitoring
+
+# Exit the script if user workload monitoring pods are not in a `running` state after the preceding 60 second sleep:
+for PODSTATUS in $(oc get pods -n openshift-user-workload-monitoring | grep -v '^NAME' | awk '{print $3}')
+do
+  if [[ "${PODSTATUS}" != 'Running' ]]; then
+	  echo "OpenShift user workload monitoring Pods not running as expected after 60 seconds. Exiting."
+	  exit 0
+  fi
+done
```

Comment:
I think it would be nicer to put a check for running pods in a blocking loop. Sth like:
```
echo -n "Waiting for OpenShift user workload monitoring Pods."
sleep 30
while [ "$(oc get pods -n openshift-user-workload-monitoring --field-selector=status.phase!=Running | wc -l)" -gt 1 ]; do
  echo -n "."
  sleep 10
done
echo
```

Reply:
Thanks Pawel, great idea. I have done this now. I noticed that `prometheus-operator` could be `Running` before the other user workload monitoring Pods had been created, which would drop the script out of the `while` loop before all of the required Pods were running. So, I included a few `||` or conditions in the `while` loop that check to ensure that at least one of each type of user workload monitoring Pod has been created, in addition to the requirement for them to be `Running`.

Reply:
Also this has enabled me to remove the long sleep before the loop.